R/sablefish_om.r

```diff
@@ -295,26 +304,48 @@ for(y in 1:(nyears-1)){
             mort = dp.y$mort[,,1,],
             mat = dp.y$mat[,,1,],
             waa = dp.y$waa[,,1,],
-            sel =  joint_sel,
+            sel =  joint_self,
             ret = joint_ret,
             F = F40,
             avg_rec = mean(recruitment/2)
         )
+        
         ssb <- apply(out_vars$naa_tmp[,,1,]*dp.y$waa[,,1,,drop=FALSE]*dp.y$mat[,,1,,drop=FALSE], 1, sum)
-        hcr_F <- npfmc_tier3a_F(ssb/B40, F35)
-
-        TACs[y+1] <- apply(out_vars$naa_tmp[,,1,]*dp.y$waa[,,1,]*dp.y$mat[,,1,]*joint_sel, 1, sum)*afscOM::F_to_mu(hcr_F)
+        hcr_F[y] <- npfmc_tier3_F(ssb, B40, F40, 0.05)
+        
+        # calculate quantities to get TAC
+        # Project population forward by 1 year with terminal F
+        proj_N <- array(NA, dim = c(dim(out_vars$naa_tmp)))
+        jointsel <- rbind(joint_self, joint_selm)
+    
+        for(s in 1:nsexes) {
+          for(a in 1:(nages - 1)) {
+            if(a == 1) proj_N[,1,s,] <- mean(recruitment) * 0.5 # using this for now, but assessment uses 1979 - terminal
+            else proj_N[,a,s,] <- out_vars$naa_tmp[,a,s,] * exp((out_f[y] * jointsel[s,a]) + M)
+          } # end a
+          proj_N[1,nages,s,1] <- out_vars$naa_tmp[1,nages-1,s,1]*exp(-(out_f[y] * jointsel[s,nages-1]) + M) +
+                                 out_vars$naa_tmp[1,nages,s,1]*exp(-(out_f[y] * jointsel[s,nages]) + M)
+          # Now calculate TACs
+          TAC_tmp <- 0
+          FAA_tmp <- (hcr_F[y] * jointsel[s,])
+          ZAA_tmp <- FAA_tmp + M
+          TAC_tmp <- sum(FAA_tmp / ZAA_tmp * proj_N[,,s,] * (1 - exp(-ZAA_tmp)) * dp.y$waa[,,s,])
```

Comment:
This is the baranov equation right? If so, let's call `baranov()` directly here.